plugins/ingestion-kafka/src/internalClusterTest/java/org/opensearch/plugin/kafka/IngestFromKafkaIT.java

```diff
@@ -236,4 +250,320 @@ public void testMultiThreadedWrites() throws Exception {
+                && Arrays.stream(ingestionState.getShardStates())
+                    .allMatch(state -> state.isPollerPaused() && state.pollerState().equalsIgnoreCase("paused"));
+        });
+
+        for (int i = 10; i < 20; i++) {
+            produceData(Integer.toString(i), "name" + i, "30");
+        }
+
+        // replica must not ingest when paused
+        Thread.sleep(1000);
+        RangeQueryBuilder rangeQueryBuilder = new RangeQueryBuilder("age").gte(28);
+        SearchResponse replicaResponse = client(nodeB).prepareSearch(indexName)
+            .setPreference("_only_local")
+            .setQuery(rangeQueryBuilder)
+            .get();
+        assertEquals(10, replicaResponse.getHits().getTotalHits().value());
+
+        // resume ingestion
+        ResumeIngestionResponse resumeResponse = resumeIngestion(indexName);
+        assertTrue(resumeResponse.isAcknowledged());
+        assertTrue(resumeResponse.isShardsAcknowledged());
+        waitForState(() -> {
+            GetIngestionStateResponse ingestionState = getIngestionState(indexName);
+            return ingestionState.getShardStates().length == 2
+                && Arrays.stream(ingestionState.getShardStates())
+                    .allMatch(
+                        state -> state.isPollerPaused() == false
+                            && (state.pollerState().equalsIgnoreCase("polling") || state.pollerState().equalsIgnoreCase("processing"))
+                    );
+        });
+
+        // verify replica ingests data after resuming ingestion
+        waitForState(() -> {
+            RangeQueryBuilder query = new RangeQueryBuilder("age").gte(28);
+            SearchResponse responseA = client(nodeA).prepareSearch(indexName).setPreference("_only_local").setQuery(query).get();
+            SearchResponse responseB = client(nodeB).prepareSearch(indexName).setPreference("_only_local").setQuery(query).get();
+            return responseA.getHits().getTotalHits().value() == 20 && responseB.getHits().getTotalHits().value() == 20;
+        });
+
+        // produce 10 more messages
+        for (int i = 20; i < 30; i++) {
+            produceData(Integer.toString(i), "name" + i, "30");
+        }
+
+        // Add new node and wait for new node to join cluster
+        final String nodeC = internalCluster().startDataOnlyNode();
+        assertBusy(() -> {
+            assertEquals("Should have 4 nodes total (1 master + 3 data)", 4, internalCluster().clusterService().state().nodes().getSize());
+        }, 30, TimeUnit.SECONDS);
+
+        // move replica from nodeB to nodeC
+        ensureGreen(indexName);
+        client().admin().cluster().prepareReroute().add(new MoveAllocationCommand(indexName, 0, nodeB, nodeC)).get();
+        ensureGreen(indexName);
+
+        // confirm replica ingests messages after moving to new node
+        waitForState(() -> {
+            RangeQueryBuilder query = new RangeQueryBuilder("age").gte(28);
```

Comment:
a bit nitpicky, but you can use `getSearchableDocCount` in `KafkaIngestionBaseIT` instead of reconstructing a range query every time.

```java
return getSearchableDocCount(nodeA) == 30 && getSearchableDocCount(nodeC) == 30;
```